docs/components/onclick.md

```diff
@@ -1,3 +1,28 @@
 # matestack core component: Action
 
-Show [specs](../../spec/usage/components/action_spec.rb)
+Show [specs](../../spec/usage/components/onclick_spec.rb)
+
+The `onclick` component renders a div that runs a function on click. This is a nice component for use cases where you would like to wrap multiple components into one click function. The function called by the onclick component requires two fields. These two fields are `emit` and `data`. These two fields are then emitted as a event in javascript when the component is clicked
+
+```ruby
+class Pages::MyPage::Home < Matestack::Ui::Page
+  def response
+    components{
+      onclick run_this_function do
+        plain "Hello world"
+      end
+    }
+  end
+
+  def run_this_function
```

Comment:
I think this is a bit misleading: the ruby method `run_this_function` just returns a config hash, which determines the client side behavior of the `onclick` component. I would rather call the ruby method `onclick_config` although it sounds not as nice as yours. I just don't want to confuse the reader.